editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -1985,9 +1985,10 @@ void Node3DEditorViewport::_sinput(const Ref<InputEvent> &p_event) {
 			} else {
 				const bool movement_threshold_passed = _edit.original_mouse_pos.distance_to(_edit.mouse_pos) > 8 * EDSCALE;
 
-				// enable region-select if nothing has been selected yet or multi-select (shift key) is active
-				if (selection_in_progress && movement_threshold_passed) {
-					if (get_selected_count() == 0 || clicked_wants_append) {
+				if (selection_in_progress && movement_threshold_passed && clicked.is_valid()) {
+					bool node_in_selection_clicked = editor_selection->is_selected(Object::cast_to<Node>(ObjectDB::get_instance(clicked)));
+
+					if (!node_in_selection_clicked) {
```

Comment:
```suggestion
					if (!node_in_selection_clicked || clicked_wants_append) {
```
Maybe keep the shift behaviour, haven't tested but was in the original

Reply:
This is a really good point and would have been a regression, good catch.